src/hibernate-resume/hibernate-resume-generator.c

```diff
@@ -54,6 +58,58 @@ static int parse_proc_cmdline_item(const char *key, const char *value, void *dat
         return 0;
 }
 
+static int add_dissected_swap_cryptsetup(void) {
+
+#if HAVE_LIBCRYPTSETUP
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *d = NULL;
+        int r;
+
```

Comment:
so if i get this right, then this generates pretty much the same service unit file that gpt-auto-generator would generate after the initrd transition, but this code here does it before, right?

can you add a comment about that, making that clear?